temporalio/contrib/openai_agents/temporal_tools.py

```diff
@@ -1,17 +1,21 @@
 """Support for using Temporal activities as OpenAI agents tools."""
 
+import json
 from datetime import timedelta
 from typing import Any, Callable, Optional
 
 from temporalio import activity, workflow
 from temporalio.common import Priority, RetryPolicy
-from temporalio.exceptions import ApplicationError
+from temporalio.exceptions import ApplicationError, FailureError
 from temporalio.workflow import ActivityCancellationType, VersioningIntent, unsafe
 
 with unsafe.imports_passed_through():
     from agents import FunctionTool, RunContextWrapper, Tool
     from agents.function_schema import function_schema
 
+class ToolSerializationError(FailureError):
```

Comment:
This should extend `TemporalError` IMO. Only errors with proto representations should extend `FailureError`.